restapi.go

```diff
@@ -2339,3 +2361,215 @@ func (s *Session) RelationshipsMutualGet(userID string) (mf []*User, err error)
 	err = unmarshal(body, &mf)
 	return
 }
+
+// ------------------------------------------------------------------------------------------------
+// Functions specific to application (slash) commands
+// ------------------------------------------------------------------------------------------------
+
+// ApplicationCommandCreate creates an global application command and returns it.
+// appID       : The application ID. If empty - s.State.User.ID. Note: works only with opened session.
+// guildID     : Guild ID to create guild-specific application command. If empty - creates global appication command.
+// cmd         : New application command data.
+func (s *Session) ApplicationCommandCreate(appID string, guildID string, cmd *ApplicationCommand) (ccmd *ApplicationCommand, err error) {
+	if appID == "" {
+		appID = s.State.User.ID
+	}
+
+	endpoint := EndpointApplicationGlobalCommands(appID)
+	if guildID != "" {
+		endpoint = EndpointApplicationGuildCommands(appID, guildID)
+	}
+
+	var body []byte
+	body, err = s.RequestWithBucketID("POST", endpoint, *cmd, endpoint)
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &ccmd)
```

Comment:
```
rcmd := &ApplicationCommand{}
err := unmarshal(body, rcmd)
if err != nil {
    return nil, err
}
return rcmd, nil
```

Reply:
First of all... there's no `rcmd` variable.

Reply:
rcmd - responseCMD, or resultCMD

Reply:
I don't know why in many similar functions u have different names for result variable

Reply:
Because in one function variable means "old command" in other "received" and "created".